### PR TITLE
docs(compiler): explain `compilerPath` argument and related fixes

### PR DESCRIPTION
**Pull request by davidyuk**
This PR is supported by the Æternity Crypto Foundation

### CODE REVIEW

File: src/contract/compiler/Cli.ts [hunk 1]
```diff
@@ -17,14 +17,19 @@ const getPackagePath = (): string => {
...
+ * A wrapper around aesophia_cli, available only in Node.js.
+ * Requires Erlang installed, assumes that `escript` is available in PATH.
  */
 export default class CompilerCli extends CompilerBase {
   #path: string;
 
   #ensureCompatibleVersion = Promise.resolve();
 
+  /**
+   * @param compilerPath - A path to aesophia_cli binary, by default uses the integrated one
```
Comment (marc0olo) [inline src/contract/compiler/Cli.ts:29]: "integrated one" means that the CLI is actually shipped along with the SDK. do I understand correctly?

Comment (marc0olo) [reply src/contract/compiler/Cli.ts:29]: I assume that windows is not supported here using the CLI compiler, right? 🤔

**Comment by marc0olo** [comment]
it would also be great to have this explained in the contract guide here:
- https://github.com/aeternity/aepp-sdk-js/blob/develop/docs/guides/contracts.md

few things I would love to see added/adjusted:

1. explain the difference between CLI compiler and http compiler
    - also, provide a note that the "hosted" compiler will be decommissioned and tell the user to run their own local http compiler if needed or to install the erlang dependency to use the shipped CLI
    - also provide a note that windows doesn't support the CLI compiler (if that's the case)
1. show usage of compiler methods directly that show how to:
    - generate & use bytecode & ACI (seems currently missing)
1. refer to or explain the utility method to get the filesystem map

File: src/contract/compiler/Cli.ts (hunk 1 above)
Comment (marc0olo) [reply src/contract/compiler/Cli.ts:29]: that's cool. wouldn't have expected the CLI binary of the compiler to work with windows

Comment (davidyuk) [reply src/contract/compiler/Cli.ts:29]: it is an Erlang VM binary

**Pull request merged**